GitOut/Features/Collections/SortedLazyAsyncCollection.cs

```diff
@@ -1,29 +1,30 @@
 using System;
 using System.Collections.Generic;
 using System.Threading.Tasks;
+using GitOut.Features.IO;
 
 namespace GitOut.Features.Collections
 {
     public class SortedLazyAsyncCollection<T> : SortedObservableCollection<T>, ILazyAsyncEnumerable<T>
     {
-        private readonly Func<IAsyncEnumerable<T>> factory;
+        private readonly Func<RelativeDirectoryPath, IAsyncEnumerable<T>> factory;
         private bool isMaterialized;
 
         public SortedLazyAsyncCollection(
-            Func<IAsyncEnumerable<T>> factory,
+            Func<RelativeDirectoryPath, IAsyncEnumerable<T>> factory,
```

Comment:
Maybe we can make this a generic type as well, like a `TContext`, instead of binding it specifically to `RelativeDirectoryPath`.
That way we can keep this collection free from using other features and allow it to be reused. Not sure what we should use if there is no context, if the need arises I guess we can add an overloaded method or class for that .